cdap-ui/app/cdap/components/AbstractWidget/HierarchyWidget/__tests__/HierarchyWidget.tests.ts

```diff
@@ -0,0 +1,419 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { typeHandler, flatToTree, addPathToFlatMap, toJsonHandler } from '../dataHandler';
+
+describe('Unit tests for Hierarchy Widget', () => {
+  it('Should handle boolean type', () => {
+    const field = {
+      name: 'Id',
+      type: 'boolean',
+    };
+    const output = {
+      name: 'Id',
+      type: 'boolean',
+      children: [],
+    };
+    const result = typeHandler(field);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(output));
+  });
+
+  it('Should handle bytes type', () => {
+    const field = {
+      name: 'Id',
+      type: 'bytes',
+    };
+    const output = {
+      name: 'Id',
+      type: 'bytes',
+      children: [],
+    };
+    const result = typeHandler(field);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(output));
+  });
+
+  it('Should handle double type', () => {
+    const field = {
+      name: 'Id',
+      type: 'double',
+    };
+    const output = {
+      name: 'Id',
+      type: 'double',
+      children: [],
+    };
+    const result = typeHandler(field);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(output));
+  });
+
+  it('Should handle float type', () => {
+    const field = {
+      name: 'Id',
+      type: 'float',
+    };
+    const output = {
+      name: 'Id',
+      type: 'float',
+      children: [],
+    };
+    const result = typeHandler(field);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(output));
+  });
+
+  it('Should handle int type', () => {
+    const field = {
+      name: 'Id',
+      type: 'int',
+    };
+    const output = {
+      name: 'Id',
+      type: 'int',
+      children: [],
+    };
+    const result = typeHandler(field);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(output));
+  });
+
+  it('Should handle long type', () => {
+    const field = {
+      name: 'Id',
+      type: 'long',
+    };
+    const output = {
+      name: 'Id',
+      type: 'long',
+      children: [],
+    };
+    const result = typeHandler(field);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(output));
+  });
+
+  it('Should handle string type', () => {
+    const field = {
+      name: 'Id',
+      type: 'string',
+    };
+    const output = {
+      name: 'Id',
+      type: 'string',
+      children: [],
+    };
+    const result = typeHandler(field);
+    expect(JSON.stringify(result)).toBe(JSON.stringify(output));
+  });
+
+  it('Should handle array type', () => {
+    const field = {
+      name: 'Id',
+      type: {
+        type: 'array',
+        items: 'string',
+      },
+      children: ['string'],
```

Comment:
The input field will not have a children key right? The avro schema spec defines only `name` and `type`?